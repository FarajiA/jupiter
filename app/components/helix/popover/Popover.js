import React from 'react';
import PropTypes from 'prop-types';
import Body from './Body';
import Footer from './Footer';
import Header from './Header';

export class Popover extends React.Component {
  componentDidMount() {
    this.checkOpen();
  }

  checkOpen = () => {
    if (this.hxPopover) {
      if (this.props.isOpen) {
        this.hxPopover.setAttribute('open', true);
      } else {
        this.hxPopover.removeAttribute('open');
      }
    }
  };

  setRef = (element) => { this.hxPopover = element; };

  render() {
    this.checkOpen();
    const { title, id, children, product, classNames, error, touched, errMsg } = this.props;
    return (
      <div className="popover">
        <div className="hxRow">
          <div className="hxCol hxSpan-4 hxOffset-1">
            <span className={`InputField-label ${classNames}`}>{title}:</span>
          </div>
          <div className="hxCol hxSpan-3 product">
            {product}
          </div>
          <div className="hxCol hxSpan-1">
            <hx-disclosure aria-controls={id} class="hxBtn hxPrimary">
              <hx-icon type="pencil" />
            </hx-disclosure>
            <hx-popover
              id={id}
              position="right-middle"
              ref={this.setRef}
            >
              {children}
            </hx-popover>
          </div>
          {touched && error && (
            <div className="hxCol hxSpan-3">
              <hx-error>
                <small>{errMsg}</small>
              </hx-error>
            </div>
          )}
        </div>
      </div>
    );
  }
}

Popover.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  isOpen: PropTypes.bool,
  product: PropTypes.string,
  classNames: PropTypes.string,
  error: PropTypes.bool,
  errMsg: PropTypes.string,
  touched: PropTypes.bool
};


Popover.Header = Header;
Popover.Body = Body;
Popover.Footer = Footer;

export default Popover;
