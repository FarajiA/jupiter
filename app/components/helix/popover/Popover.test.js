import React from 'react';
import renderer from 'react-test-renderer';
import Popover from './Popover';

describe('Popover', () => {
  const defaultProps = {
    title: 'Role',
    role: 'Aviator',
    id: 'default-id',
    children: '<h1>Popover Content</h1>'
  };

  test('it renders correct props', () => {
    const component = renderer.create(<Popover {...defaultProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders child content', () => {
    const popover = shallow(<Popover {...defaultProps} isOpen />).find('hx-popover');
    expect(popover.text()).toEqual('<h1>Popover Content</h1>');
  });

  test('it sets popover open attribute to true if isOpen is true', () => {
    const popover = shallow(<Popover {...defaultProps} isOpen />).find('hx-popover');
    expect(popover.prop('open')).toEqual(undefined);
  });

  test('it sets popover open attribute if isOpen is true', () => {
    const popover = shallow(<Popover {...defaultProps} />).find('hx-popover');
    expect(popover.prop('open')).toBeFalsy();
  });
});
