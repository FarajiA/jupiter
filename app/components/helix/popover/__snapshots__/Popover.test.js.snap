// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover it renders correct props 1`] = `
<div
  className="popover"
>
  <div
    className="hxRow"
  >
    <div
      className="hxCol hxSpan-4"
    >
      <span
        className="InputField-label role"
      >
        Role
        :
      </span>
    </div>
    <div
      className="hxCol hxSpan-4 role"
    >
      Aviator
    </div>
    <div
      className="hxCol hxSpan-4"
    >
      <hx-disclosure
        aria-controls="default-id"
        class="hxBtn hxPrimary"
      >
        <hx-icon
          type="pencil"
        />
      </hx-disclosure>
      <hx-popover
        id="default-id"
        open={null}
        position="right-middle"
      >
        &lt;h1&gt;Popover Content&lt;/h1&gt;
      </hx-popover>
    </div>
  </div>
</div>
`;
